src/openforms/registrations/contrib/zgw_apis/api/views.py

```diff
@@ -73,6 +81,60 @@ def get_objects(self):
         return case_types
 
 
+@dataclass
+class Product:
+    url: str
+    description: str = ""
+
+
+@extend_schema_view(
+    get=extend_schema(
+        summary=_(
+            "List the available products bound to a case type within a catalogue (ZGW APIs)"
+        ),
+        parameters=[ListProductsQueryParamsSerializer],
+    ),
+)
+class ProductsListView(ListMixin[Product], APIView):
+    authentication_classes = (authentication.SessionAuthentication,)
+    permission_classes = (permissions.IsAdminUser,)
+    serializer_class = CaseTypeProductSerializer
+
+    def get_objects(self):
+        filter_serializer = ListProductsQueryParamsSerializer(
+            data=self.request.query_params
+        )
+        filter_serializer.is_valid(raise_exception=True)
+
+        catalogue_url = filter_serializer.validated_data["catalogue_url"]
+        case_type_identification = filter_serializer.validated_data[
+            "case_type_identification"
+        ]
+        today = datetime_in_amsterdam(timezone.now()).date()
+
+        products: list[Product] = []
+        with filter_serializer.get_ztc_client() as client:
+            case_type_versions = client.find_case_types(
+                catalogus=catalogue_url,
+                identification=case_type_identification,
+                valid_on=today,
+            )
+            case_type_products = (
+                resolve_case_type_products(case_type_versions)
+                if case_type_versions
+                else []
+            )
+            for case_type_product in case_type_products:
+                products.append(
+                    Product(
+                        url=case_type_product["url"],
+                        description=case_type_product["description"],
```

Comment:
Can you address this type checker error? I would opt to make the key required in the product, if it's missing from the upstream API, you can default it to an empty string.

Reply:
Mhh, yeah making it required is also an option..
I was thinking `description=case_type_product.get("description", "")`